src/EventPayload.ts

```diff
@@ -71,22 +79,43 @@ export interface EventPayload {
   internalUser?: boolean;
   /** The IP address for the event.*/
   ip?: {
-    /** The IPv4 address associated with the event. */
+    /** The IPv4 address associated with the event.
+     *
+     * If not specified the system will use the Source IP address
+     * from the IP header of the request.
+     */
     address: string;
     /** The algorithm to use to anonymize the IP address after collection. */
     algorithm: string;
   };
   /** A label assigned to the event, e.g. a CTA label. */
   label?: string;
-  /** The locale of the user who generated the event. */
+  /** The locale of the user who generated the event.
+   *
+   * If not specified the system will automatically
+   * determine the locale from the Accept-Language header.
+   */
   locale?: string;
   /** The location information of the visitor for the event.
    * Either a Coordinates object with both latitude and longitude or a string
-   * with the country of the visitor for the event, as a ISO 3166-1 alpha-2 country code. */
+   * with the country of the visitor for the event, as a ISO 3166-1 alpha-2 country code.
+   *
+   * If not specified the system will automatically
+   * determine all location information from the request's IP address,
+   * or the value of the `ip` property if specified.
+   */
   location?: Coordinates | string;
-  /** The URL of the page where the event occurred */
+  /** The URL of the page where the event occurred.
+   *
+   * If not specified the system will automatically
+   * use the `Referer` header of the request.
```

Comment:
I would also specific here that the `Referer` header is automatically added when the request is made and is not something the user needs to provide

Reply:
Perhaps "....use the `Referer` header from the auto-generated request headers". Also, comment applies to the other changes that mentioned headers as well.